lib/rouge/lexers/idris.rb

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Idris < RegexLexer
+      title "Idris"
+      desc "The Idris programming language (idris-lang.org)"
+
+      tag 'idris'
+      aliases 'idr'
+      filenames '*.idr'
+      mimetypes 'text/x-idris'
+
+      reserved = %w(
+        _
+        data class instance namespace
+        infix[lr]? let where of with type 
+        do if then else case in
+      )
```

Comment:
Local variables defined in the scope of the class will not be garbage collected by Ruby and so have negative performance characteristics. For this reason, we try to encourage new lexers to use memoising methods instead (there are lexers in the repo that still use local variables but moving away from this is an ongoing proejct).

You can see an example in the C lexer for how to do this:

https://github.com/rouge-ruby/rouge/blob/81b36718a602d37b296301e5e9f4c8b2156b6ba2/lib/rouge/lexers/c.rb#L18-L27

The memoised value is then later referenced like this:

https://github.com/rouge-ruby/rouge/blob/81b36718a602d37b296301e5e9f4c8b2156b6ba2/lib/rouge/lexers/c.rb#L112-L127

Reply:
Addressed in https://github.com/rouge-ruby/rouge/pull/1464/commits/d62d75e39180b4c87f5f64fa122918acfa6a6fe5